src/controller/tests/data_model/TestRead.cpp

```diff
@@ -1672,12 +1698,19 @@ void TestReadInteraction::TestReadHandler_KillOverQuotaSubscriptions(nlTestSuite
 
     // The following check will trigger the logic in im to kill the read handlers that use more paths than the limit per fabric.
     {
-        EstablishSubscriptions(apSuite, ctx.GetSessionAliceToBob(),
-                               app::InteractionModelEngine::kMinSupportedSubscriptionsPerFabric,
-                               app::InteractionModelEngine::kMinSupportedPathsPerSubscription, &readCallbackFabric2, readClients);
-
-        ctx.GetIOContext().DriveIOUntil(System::Clock::Seconds16(60),
-                                        [&]() { return readCallbackFabric2.mOnSubscriptionEstablishedCount == 1; });
+        EstablishReadOrSubscriptions(
+            apSuite, ctx.GetSessionAliceToBob(), app::InteractionModelEngine::kMinSupportedSubscriptionsPerFabric,
+            app::InteractionModelEngine::kMinSupportedPathsPerSubscription,
+            app::AttributePathParams(kTestEndpointId, TestCluster::Id, TestCluster::Attributes::Int16u::Id),
+            app::ReadClient::InteractionType::Subscribe, &readCallbackFabric2, readClients);
+
+        ctx.GetIOContext().DriveIOUntil(System::Clock::Seconds16(5), [&]() {
+            return readCallbackFabric2.mOnSubscriptionEstablishedCount ==
+                app::InteractionModelEngine::kMinSupportedSubscriptionsPerFabric &&
+                readCallbackFabric2.mAttributeCount ==
+                app::InteractionModelEngine::kMinSupportedPathsPerSubscription *
+                    app::InteractionModelEngine::kMinSupportedSubscriptionsPerFabric;
```

Comment:
Shouldn't we be validating the # of subscriptions established on fabric A on the server vs. fabric B?

Reply:
This is what we validate later.